static/js/pages/RecipesPage/SearchComponent/SearchComponent.js

```diff
@@ -67,58 +67,62 @@ class SearchComponent extends Component {
         this.setState({ elements: [] });
     }
 
+    requestToSearch(item, searchParam) {
+        wait(200)
+        .then(() => {
+            axios.post(`/api/recipes/search/${searchParam}`, { item })
+                .then(response => this.setState({ process: 'fetched', elements: response.data }))
+                .catch(error => console.log(error));
+        })
+        .catch((err) => {
+            this.setState({ process: 'failedToFetch' });
+            console.log(err, 'Failed to get recipes data');
+        });
+    }
+
     elementSearch(item) {
         this.setState({ item });
         if (item) {
             switch (this.state.type) {
             case 'searchByName':
-                wait(1000)
-                .then(() => {
-                    axios.post('/api/recipes/search/name', { item })
-                        .then(response => this.setState({ process: 'fetched', elements: response.data }))
-                        .catch(error => console.log(error));
-                })
-                .catch((err) => {
-                    this.setState({ process: 'failedToFetch' });
-                    console.log(err, 'Failed to get recipes data');
-                });
+                this.requestToSearch(item, 'name');
                 break;
             case 'searchByTagCategory':
-                wait(1000)
-                .then(() => {
-                    axios.post('/api/recipes/search/tagtype', { item })
-                        .then(response => this.setState({ process: 'fetched', elements: response.data }))
-                        .catch(error => console.log(error));
-                })
-                .catch((err) => {
-                    this.setState({ process: 'failedToFetch' });
-                    console.log(err, 'Failed to get recipes data');
-                });
+                this.requestToSearch(item, 'tagtype');
                 break;
             default:
-                this.setState({ elements: [] });
+                this.setState({ item: '', elements: [], process: 'fetched' });
             }
         } else {
-            this.setState({ item: '' });
-            this.setState({ elements: [] });
-            this.setState({ process: '' });
+            this.setState({ item: '', elements: [], process: '' });
         }
     }
 
     typeChange(type) {
-        this.setState({ type });
-        this.setState({ elements: [] });
+        this.setState({ type, elements: [] });
     }
 
     render() {
         const phase = this.state.process;
+        const type = this.state.type;
         let searchElement = null;
+        let inputSearch = null;
```

Comment:
what is the difference between searchElement and inputSearch and why we need 2 variables?

Reply:
two different components are displayed at the same time. There can't be one variable